### PR TITLE
fix: prevent deadlock on single-core systems and with empty arrays in `utils/parallel`

### PR DESCRIPTION
**Pull request by shiavm006**
## Description

> What is the purpose of this pull request?

This pull request:

fixes a deadlock bug in `@stdlib/utils/parallel` where the function would hang indefinitely on single-core systems (`numCPUs = 1`) because defaults would set `workers = 0`. a deadlock bug when called with empty arrays (`parallel([])`) where clamping logic would reduce workers to 0.
fixes a deadlock bug when explicitly passing `{workers: 0, concurrency: 0}` options and updated `defaults.js` to use `@stdlib/math/base/special/fast/max` to ensure minimum of 1 worker/concurrency as suggested by maintainer ,adds safeguard in `main.js` to prevent workers/concurrency from dropping below 1 after clamping logic , tests covering all three edge cases.

## Related Issues

> Does this pull request have any related issues?

This pull request has the following related issues:

-   No related issue (discussed with maintainer @kgryte via direct message)

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

This fix was discussed with maintainer @kgryte who suggested using `@stdlib/math/base/special/fast/max` for the defaults.js fix. The safeguard in main.js was added to handle edge cases where clamping could still result in 0 workers even with the defaults fix.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

### AI Assistance

> When authoring the changes proposed in this PR, did you use any kind of AI assistance?

-   [ ] Yes
-   [x] No

If you answered "yes" above, how did you use AI assistance?

-   [ ] Code generation (e.g., when writing an implementation or fixing a bug)
-   [ ] Test/benchmark generation
-   [ ] Documentation (including examples)
-   [ ] Research and understanding

N/A

* * *
@stdlib-js/reviewers
[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by shiavm006** [comment]
cc @kgryte i verified it locally ya u can check it out like can disscuss on changes and implement that same

**Comment by shiavm006** [comment]
@kgryte what u think about it

File: lib/node_modules/@stdlib/utils/parallel/lib/defaults.js [hunk 1]
```diff
@@ -36,12 +37,17 @@ var numCPUs = require( '@stdlib/os/num-cpus' );
 * // returns {...}
 */
 function defaults() {
+	var n;
+
+	// Ensure at least 1 worker and 1 concurrent script to prevent deadlock on single-core systems:
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/parallel/lib/defaults.js]:
```suggestion
	// Accommodate single-core systems by ensuring at least 1 worker and 1 concurrent script:
```

File: lib/node_modules/@stdlib/utils/parallel/lib/main.js [hunk 2]
```diff
@@ -102,6 +102,13 @@ function parallel() {
 	if ( opts.workers > opts.concurrency ) {
 		opts.workers = opts.concurrency;
 	}
+	// Ensure at least 1 worker and 1 concurrent script to prevent deadlock (even with empty arrays or single-core systems):
+	if ( opts.workers < 1 ) {
+		opts.workers = 1;
+	}
+	if ( opts.concurrency < 1 ) {
+		opts.concurrency = 1;
+	}
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/parallel/lib/main.js]:
```suggestion
```

This isn't necessary. L83 instantiates defaults above which are already clamped to never be less than unity.

Comment (kgryte) [reply lib/node_modules/@stdlib/utils/parallel/lib/main.js]: Oh, I see, because we reset the concurrency for an empty file array.

File: lib/node_modules/@stdlib/utils/parallel/test/test.main.js [hunk 3]
```diff
@@ -305,6 +305,45 @@ tape( 'if the number of workers is greater than the concurrency, the function se
 	}
 });
 
+tape( 'the function ensures at least 1 worker and 1 concurrent script on single-core systems', function test( t ) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/parallel/test/test.main.js]:
```suggestion
tape( 'the function ensures at least one worker and a concurrency greater than or equal to unity', function test( t ) {
```

File: lib/node_modules/@stdlib/utils/parallel/test/test.main.js [hunk 4]
```diff
@@ -305,6 +305,45 @@ tape( 'if the number of workers is greater than the concurrency, the function se
...
+			'encoding': 'buffer',
+			'maxBuffer': 200 * 1024 * 1024
+		};
+	}
+
+	function done( error ) {
+		if ( error ) {
+			t.ok( false, error.message );
+		} else {
+			t.ok( true, 'executes successfully on single-core systems' );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/utils/parallel/test/test.main.js]:
```suggestion
			t.ok( true, 'runs scripts successfully' );
```

**Review by kgryte — APPROVED** [review]
LGTM
